rust/connlib/tunnel/src/peer.rs

```diff
@@ -207,9 +343,447 @@ impl GatewayOnClient {
     }
 }
 
+struct GatewayResource {
+    ip: IpNetwork,
+    filters: Filters,
+    expires_at: Option<DateTime<Utc>>,
+}
+
 /// The state of one client on a gateway.
 pub struct ClientOnGateway {
     id: ClientId,
     allowed_ips: IpNetworkTable<()>,
-    resources: IpNetworkTable<ExpiryingResource>,
+    resources: HashMap<ResourceId, GatewayResource>,
+    filters: IpNetworkTable<FilterEngine>,
+}
+
+#[cfg(test)]
+mod tests {
```

Comment:
#4791

Reply:
It would also be nice to add filtering tests to the state-machine proptests but that's a different story

Reply:
> #4791

Can we just include the proptests in this PR? If we already know that the coverage of these tests isn't good enough then let's not delay that.

It would be nice to ever so slightly increase the baseline for what we consider good enough to merge.

Reply:
> It would also be nice to add filtering tests to the state-machine proptests but that's a different story

Yes but they aren't ready for that unfortunately.